docs/changelog.md

```diff
@@ -29,6 +29,10 @@ permalink: /changelog/
      information.
    * A method for sending metrics on custom pings has been added. See
      `docs/pings/custom.md` for more information.
+   * Disabling telemetry through `setUploadEnabled` now clears all metrics
+     immediately.
+   * ⚠️ **This is a breaking API change**: `Glean.setUploadEnabled` now requires
+     an application context to be passed in.
```

Comment:
This is unfortunate, but I haven't been able to find a better solution.  The release linter says we can't store a reference to the applicationContext in the Glean singleton since it would leak the application (which is what I did in the initial version of this PR).

Reply:
This is ok, that's what we do in other singletons as well. The `applicationContext` is a special context that lives throughout the whole application, as our `Glean` object. They have the same lifetime, so we don't need to care :)